src/dfcx_scrapi/builders/intent.py

```diff
@@ -14,19 +14,467 @@
+            all((isinstance(p, str) for p in phrase)) and
+            all((isinstance(a, str) for a in annotations))
+        ):
+            raise ValueError(
+                "Only strings allowed in phrase or annotations list."
+            )
+        if len(annotations) > len(phrase):
+            raise IndexError(
+                "Length of annotations list is more than phrase list!"
+            )
+
+        # Propagate the annotations list if needed
+        if len(annotations) < len(phrase):
+            annotations.extend([""] * (len(phrase) - len(annotations)))
+        # Creating parts for the training phrase
+        parts_list = []
+        for text, parameter_id in zip(phrase, annotations):
+            part = Intent.TrainingPhrase.Part(
+                text=text, parameter_id=parameter_id
+            )
+            parts_list.append(part)
+
+        # Create the training phrase obj and add it to the others
+        tp = Intent.TrainingPhrase(
+            parts=parts_list, repeat_count=repeat_count
+        )
+        self.proto_obj.training_phrases.append(tp)
+
+        return self.proto_obj
+
+
+    def add_parameter(
+        self,
+        parameter_id: str,
+        entity_type: str,
+        is_list: bool = False,
+        redact: bool = False
+    ) -> Intent:
+        """Add a parameter to Parameter attribute of proto_obj.
+
+        Args:
+          id (str):
+            Required. The unique identifier of the parameter.
+          entity_type (str):
+            Required. The entity type of the parameter. Format:
+            ``projects/-/locations/-/agents/-/
+              entityTypes/<System Entity Type ID>``
+            for system entity types (for example,
+            ``projects/-/locations/-/agents/-/entityTypes/sys.date``),
+            or
+            ``projects/<Project ID>/locations/<Location ID>/
+              agents/<Agent ID>/entityTypes/<Entity Type ID>``
+            for developer entity types.
+          is_list (bool):
+            Indicates whether the parameter represents a
+            list of values.
+          redact (bool):
+            Indicates whether the parameter content should be redacted
+            in log. If redaction is enabled, the parameter content will
+            be replaced by parameter name during logging.
+
+        Returns:
+          An Intent object stored in proto_obj
+        """
+        self._check_intent_exist()
+
+        if not (isinstance(parameter_id, str) or isinstance(entity_type, str)):
+            raise ValueError(
+                "parameter_id and entity_type should be string."
+            )
+
+        # Create the new parameter and add it to the proto_obj
+        parameters = Intent.Parameter(
+            id=parameter_id,
+            entity_type=entity_type,
+            is_list=is_list,
+            redact=redact
+        )
+        self.proto_obj.parameters.append(parameters)
+
+        return self.proto_obj
+
+
+    def add_label(
```

Comment:
@MRyderOC we'll need to add some logic/error handling or mutations to this method.

I tried to add the Dict label `{'lob':'Accounting'}` and I received this error from the base API:
```
Label value 'Accounting' violates format constraints. The value can only contain lowercase letters, numeric characters, underscores and dashes. The value can be at most 63 characters long. International characters are allowed."
```

So I think we should go ahead and enforce this constraint check inside the builder so the user doesn't spend all this time building the Intent, only for it to fail when they try to push it to their Agent.

Reply:
And on that same note, should we consider adding methods for removing Labels / Training Phrases / etc.?

In this example, I now have a label `lob:Accounting` that I can't get rid of in my Intent Proto very "easily". Technically, I can modify the proto object directly and clear out `ib.proto_obj.labels` but that sort of defeats the purpose of the easy-to-use methods.

WDYT?